File: CSharpTest/ExceptionTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.IO;

namespace CSharpTest
{
    [TestClass]
    public class ExceptionTest
    {
        [TestMethod]
        [ExpectedException(typeof(IOException), "A non-existent file was inappropriately allowed.")]
        public void CatchExceptionTest()
        {
            var f = new MyLibrary.FileIOExample();
            f.OpenFile("This file does not exist"); // We expect this to throw IOException
        }
    }
}

File: JavaTest/ExceptionTest.java
import MyLibrary.*;
import org.junit.*;

import java.io.IOException;

import static org.junit.Assert.*;

public class ExceptionTest extends TestBase {
    @Test(expected=java.io.IOException.class)
    public void CatchExceptionTest() throws java.io.IOException {
        FileIOExample f = new FileIOExample();
        f.OpenFile("This file does not exist"); // We expect this to throw java.io.IOException
    }
}
